components/api/app/controllers/api/data_sources_controller.rb

```
@@ -41,7 +41,7 @@ def update
     end
 
     def destroy
-      Authorization::Authority.authorize! :destroy, @data_source, current_user, {:or => :current_user_is_object_owner}
+      authorize! :destroy, @data_source, current_user, {:or => :current_user_is_object_owner}
       @data_source.destroy
       head :ok
     end
```

I see what you're saying with regards to the is_admin? part of the functionality. But the other part of the boolean is core to the minimal behavior of a disabled data source: `@data_source.disabled? && @data_source.owner != current_user`. 

Note that this isn't filtering the disabled data sources from the index (they are not filtered by default) , this is hiding the contents of a data source when it's disabled, so that an arbitrary user can't browse a disabled data source. This functionality should be present no matter what authorization system is in place, if any. Let's try to figure out how this should change later